renku/command/mergetool.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017-2022 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Custom git mergetool for renku metadata."""
+
+from pathlib import Path
+
+from renku.command.command_builder import inject
+from renku.command.command_builder.command import Command
+from renku.core.interface.client_dispatcher import IClientDispatcher
+
+
+def mergetool_command():
+    """Command to move or rename a file, a directory, or a symlink."""
+    return Command().command(_mergetool).require_migration().with_database()
+
+
+@inject.autoparams()
+def _mergetool(local: Path, remote: Path, base: Path) -> None:
```

Comment:
We should put this and `setup_mergetool` in `core`.

Reply:
It depends directly on GitMerger which is an infrastructure thing. so we'd have to either create an interface for this, create a command builder extension to inject GitMerger, depend on the interface here and then call the command with that. Or just leave it here and then this depending directly on infrastructure is fine.

I decided to do the lazy thing, as the implementation and concept of a git merge tool is so git specific that having an interface just for the sake of the flow of dependencies seemed overkill.